protocols/identify/src/handler.rs

```diff
@@ -52,11 +55,22 @@ pub struct Handler {
     inbound_identify_push: Option<BoxFuture<'static, Result<PushInfo, UpgradeError>>>,
     /// Pending events to yield.
     events: SmallVec<
-        [ConnectionHandlerEvent<Either<Identify, Push<OutboundPush>>, (), Event, io::Error>; 4],
+        [ConnectionHandlerEvent<
+            Either<ReadyUpgrade<StreamProtocol>, ReadyUpgrade<StreamProtocol>>,
+            (),
+            Event,
+            io::Error,
+        >; 4],
     >,
 
     /// Pending identification replies, awaiting being sent.
-    pending_replies: FuturesUnordered<BoxFuture<'static, Result<(), UpgradeError>>>,
+    pending_replies: futures_bounded::FuturesSet<Result<(), UpgradeError>>,
+
+    /// Pending identify requests.
+    outbound_identify_futs: futures_bounded::FuturesSet<Result<Info, UpgradeError>>,
+
+    /// Pending identify/push requests.
+    outbound_identify_push_futs: futures_bounded::FuturesSet<Result<(), UpgradeError>>,
```

Comment:
Can we merge these somehow and still retain all the different branches of behaviour in `poll`? That would be great because we wouldn't have to `poll` 3 different `FuturesSet`s.